ext/mysql2/result.h

```diff
@@ -1,5 +1,6 @@
 #ifndef MYSQL2_RESULT_H
 #define MYSQL2_RESULT_H
+#include <stdbool.h>
```

Comment:
I also found that `result.c` is already using `uintptr_t` which is a C99 type from `stdint.h` but we've not included `stdint.h` ourselves, which is probably an error.